### PR TITLE
Configure colour for status messages #4080

### LINKED ISSUE #4080: Configure colour for status messages
Have an option to specify the colour of the status message in status message tags
Info (light blue) will be the default colour. Use yellow for warnings, red for errors, green for success.

### PR DESCRIPTION
**Pull request by taniach**
Fixes #4080

### CODE REVIEW

File: src/main/java/teammates/common/util/StatusMessageColors.java [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+                    return "success";
+                case WARNING:
+                    return "warning";
+                case DANGER:
+                    return "danger";
+                case INFO:
+                default:
+                    return "info";
+            }
+        } catch (Exception e) {
```
Comment (damithc) [inline src/main/java/teammates/common/util/StatusMessageColors.java]: Swallowing a totally unexpected exception and pretending it never happened is not good. There should be an assertion here.

File: src/main/java/teammates/common/util/StringHelper.java [hunk 2]
```diff
@@ -483,4 +485,49 @@ public static String convertToEmptyStringIfNull(String str) {
         return (str == null) ? "" : str;
     }
 
+    public static String removeVariables(String statusMessage) {
```
Comment (damithc) [inline src/main/java/teammates/common/util/StringHelper.java]: What's this for? Public methods should have a header comment unless the method is trivial.

File: src/main/java/teammates/common/util/StatusMessageColors.java [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            return "info";
+        }
+    }
+
```
Comment (damithc) [inline src/main/java/teammates/common/util/StatusMessageColors.java]: Consider creating a StatusMessage class that encapsulate color, text etc. If not in this PR, as a separate PR.

File: src/main/java/teammates/common/util/StatusMessageColors.java (hunk 1 above)
Comment (taniach) [inline src/main/java/teammates/common/util/StatusMessageColors.java]: This is for exceptions which occur when the status message cannot be found in the HashMap e.g. the colour has not been specified in the Map. In that case the status message is `info`

File: src/main/java/teammates/common/util/StringHelper.java (hunk 2 above)
Comment (taniach) [inline src/main/java/teammates/common/util/StringHelper.java]: Some status messages in `Const.java` have templates in them: `You are enrolled in the course %s`. This is the message stored in the HashMap. The actual status message has the course ID in place of `%s`. In order to search for the message in the HashMap, I called the method for the template message and the actual message to remove `%s` and the actual course ID respectively.

File: src/main/java/teammates/common/util/StatusMessageColors.java (hunk 3 above)
Comment (taniach) [inline src/main/java/teammates/common/util/StatusMessageColors.java]: Created #4119

File: src/main/java/teammates/common/util/StatusMessageColors.java (hunk 1 above)
Comment (damithc) [inline src/main/java/teammates/common/util/StatusMessageColors.java]: Should not happen right? In that case use an assertion. GAE does not allow normal asserts. So need to use our own `Assumption` class instead.

File: src/main/java/teammates/common/util/StringHelper.java [hunk 4]
```diff
@@ -483,4 +485,59 @@ public static String convertToEmptyStringIfNull(String str) {
...
+     * Removes templates and actual values filled in place of templates
+     * for status messages
+     * 
+     * Example : "You are enrolled in %s" and "You are enrolled in CS2103"
+     *            are converted to "You are enrolled in "
+     *            
+     * @param statusMessage
+     * @return status message without variables and actual values
+     */
+    public static String removeVariables(String statusMessage) {
```
Comment (damithc) [inline src/main/java/teammates/common/util/StringHelper.java]: It's a bit hackish having to do this. You wont need it if you introduce a new `StatusMessage` class. The importance of the message can be decided at the point you instantiate the object, not afterwards.

Comment (taniach) [inline src/main/java/teammates/common/util/StringHelper.java]: Ok. I feel this is very fragile too. Any change to `Const.java` status messages may affect this and cause it to fail.

**Comment by taniach** [comment]
Created a new PR #4142 since there are many changes

**Pull request closed without merging**